save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/cli/CliRunner.kt

```diff
@@ -0,0 +1,88 @@
+package com.saveourtool.save.demo.runners.cli
+
+import com.saveourtool.save.demo.DemoAdditionalParams
+import com.saveourtool.save.demo.DemoResult
+import com.saveourtool.save.demo.runners.Runner
+import com.saveourtool.save.demo.utils.generateName
+import org.springframework.stereotype.Component
+import java.io.File
+import java.nio.file.Path
+import kotlin.io.path.div
+
+/**
+ * Interface that should be implemented by all the runners that use [ProcessBuilder] in order to run tools for demo.
+ */
+@Component
+interface CliRunner <in P : DemoAdditionalParams, out R : DemoResult> : Runner<P, R> {
+    /**
+     * @return string that starts with Linux, Windows or macOS
+     */
+    fun osName(): String = System.getProperty("os.name")
```

Comment:
why osName is interface of CliRunner? it's property of run system (save-demo). can we move it to `createProcessBuilder` as local variable